src/optional.h

```diff
@@ -234,6 +234,30 @@ class optional
         }
 };
 
+template<class T, class U>
+constexpr bool operator==( const optional<T> &lhs, const optional<U> &rhs )
+{
+    if( bool( lhs ) != bool( rhs ) ) {
```

Comment:
It probably should be
```cpp
    if( lhs.has_value() != rhs.has_value() ) {
```
Same for the `operator!=`.

Reply:
`operator bool()` works the same as `has_value()`.

Reply:
We generally don't use c-style or function-style casts, instead we use `static_cast`. But in this case `has_value` works the same.